patterns/direct_processor.py

```diff
@@ -0,0 +1,142 @@
+
+# Broker Config. Note: Could pass other properties Look into
+broker_props = {
+    "solace.messaging.transport.host": os.environ.get('SOLACE_HOST') or "tcp://localhost:55555,tcp://localhost:55554",
+    "solace.messaging.service.vpn-name": os.environ.get('SOLACE_VPN') or "default",
+    "solace.messaging.authentication.scheme.basic.username": os.environ.get('SOLACE_USERNAME') or "default",
+    "solace.messaging.authentication.scheme.basic.password": os.environ.get('SOLACE_PASSWORD') or ""
+    }
+
+# Build A messaging service with a reconnection strategy of 20 retries over an interval of 3 seconds
+# Note: The reconnections strategy could also be configured using the broker properties object
+messaging_service = MessagingService.builder().from_properties(broker_props)\
+                        .with_reconnection_retry_strategy(RetryStrategy.parametrized_retry(20,3))\
+                        .build()
+
+# Blocking connect thread
+messaging_service.connect()
+print(f'\nMessaging Service connected? {messaging_service.is_connected}')
+
+# Inner classes for connection error handling
+class ServiceEventHandler(ReconnectionListener, ReconnectionAttemptListener, ServiceInterruptionListener):
+    def on_reconnected(self, e: ServiceEvent):
+        print("\non_reconnected")
+        print(f"Error cause: {e.get_cause()}")
+        print(f"Message: {e.get_message()}")
+    
+    def on_reconnecting(self, e: "ServiceEvent"):
+        print("\non_reconnecting")
+        print(f"Error cause: {e.get_cause()}")
+        print(f"Message: {e.get_message()}")
+
+    def on_service_interrupted(self, e: "ServiceEvent"):
+        print("\non_service_interrupted")
+        print(f"Error cause: {e.get_cause()}")
+        print(f"Message: {e.get_message()}")
+
+# Event Handling for the messaging service
+service_handler = ServiceEventHandler()
+messaging_service.add_reconnection_listener(service_handler)
+messaging_service.add_reconnection_attempt_listener(service_handler)
+messaging_service.add_service_interruption_listener(service_handler)
+ 
+# Inner classes for publish error handling
+class PublisherErrorHandling(PublishFailureListener):
+    def on_failed_publish(self, e: "FailedPublishEvent"):
+        print("on_failed_publish")
+
+# Handle received messages
+class MessageHandlerImpl(MessageHandler):
+    def on_message(self, message: InboundMessage):
+        out_topic = Topic.of(TOPIC_PREFIX + f'/direct/processor/output')
+        in_topic = message.get_destination_name()
+        payload_str = message.get_payload_as_string()
+        print(f'Received input message (body string):\n{payload_str}')
+        print(f'Message dump: {message.solace_message.get_message_dump()} \n')
+
+        processed_payload_str = payload_str.upper()
+        print(f'Published output message (body):\n{processed_payload_str}')
+
+        output_msg = msg_builder \
+                        .with_application_message_id(message.get_application_message_id())\
```

Comment:
As per the [docs](https://docs.solace.com/API-Developer-Online-Ref-Documentation/python/source/rst/solace.messaging.html?#solace.messaging.message.Message.get_application_message_id), `get_application_message_id()` 
>Returns Sender application identifier if set by message publisher, or None/empty if not set.

If the sender does not set the application message ID you will get the following warning when attempting to build a message builder 
>[WARNING] solace.messaging.publisher: [_solace_utilities.py:227]  Invalid datatype. Expected type: [<class 'str'>], but actual [<class 'NoneType'>]

Reply:
Checked for get_application_message_id() is not NONE, then set it on the published message.